### PR TITLE
Sanctum implementation to auth

### PR DESCRIPTION
**Pull request by feroj21**
Implemented sanctum to get the SPA cookie based authentication feature.

### CODE REVIEW

**Comment by feroj21** [comment]
[Laravel auth middleware](https://trello.com/c/3rzruaP6/130-laravel-auth-middleware)

File: config/cors.php [hunk 1]
```diff
@@ -15,7 +15,7 @@
     |
     */
 
-    'paths' => ['api/*', 'sanctum/csrf-cookie'],
+    'paths' => ['api/*', 'sanctum/csrf-cookie', 'login', 'logout'],
```
Comment (LCD344) [inline config/cors.php:18]: I think the login and logout will also be api routes in this case?

File: app/Models/User.php [hunk 2]
```diff
@@ -6,10 +6,11 @@
 use Illuminate\Database\Eloquent\Factories\HasFactory;
 use Illuminate\Foundation\Auth\User as Authenticatable;
 use Illuminate\Notifications\Notifiable;
+use Laravel\Sanctum\HasApiTokens;
 
 class User extends Authenticatable
 {
-    use HasFactory, Notifiable;
+    use HasFactory, Notifiable, HasApiTokens;
```
Comment (LCD344) [inline app/Models/User.php:13]: do we need the HasApiToken for the SPA authentication type of sanctum

Comment (feroj21) [reply app/Models/User.php:13]: No we dont, but we need this to test using POSTMAN.

**Pull request merged**